packages/material-ui-lab/src/LoadingButton/LoadingButton.js

```diff
@@ -1,95 +1,133 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
-import clsx from 'clsx';
 import { chainPropTypes } from '@material-ui/utils';
 import { capitalize } from '@material-ui/core/utils';
-import { withStyles } from '@material-ui/core/styles';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import {
+  experimentalStyled as styled,
+  unstable_useThemeProps as useThemeProps,
+} from '@material-ui/core/styles';
 import Button from '@material-ui/core/Button';
 import CircularProgress from '@material-ui/core/CircularProgress';
+import loadingButtonClasses, { getLoadingButtonUtilityClass } from './loadingButtonClasses';
 
-export const styles = () => ({
-  /* Styles applied to the root element. */
-  root: {},
-  /* Styles applied to the root element if `loading={true}`. */
-  loading: {},
-  /* Styles applied to the loadingIndicator element. */
-  loadingIndicator: {
-    position: 'absolute',
-    visibility: 'visible',
-    display: 'flex',
-  },
-  /* Styles applied to the loadingIndicator element if `loadingPosition="center"`. */
-  loadingIndicatorCenter: {
-    left: '50%',
-    transform: 'translate(-50%)',
-  },
-  /* Styles applied to the loadingIndicator element if `loadingPosition="start"`. */
-  loadingIndicatorStart: {
-    left: 14,
+const useUtilityClasses = (styleProps) => {
+  const { loading, loadingPosition, classes } = styleProps;
+
+  const slots = {
+    root: ['root', loading && 'loading'],
+    startIcon: [`startIcon${loading ? 'Loading' : ''}${capitalize(loadingPosition)}`],
+    endIcon: [`endIcon${loading ? 'Loading' : ''}${capitalize(loadingPosition)}`],
+    label: [`label${loading ? 'Loading' : ''}${capitalize(loadingPosition)}`],
+    loadingIndicator: ['loadingIndicator', `loadingIndicator${capitalize(loadingPosition)}`],
+  };
+
+  return composeClasses(slots, getLoadingButtonUtilityClass, classes);
+};
+
+const buttonShouldForwardProp = (prop) =>
+  prop !== 'styleProps' && prop !== 'theme' && prop !== 'isRtl' && prop !== 'sx' && prop !== 'as';
+const LoadingButtonRoot = styled(
+  Button,
+  {
+    shouldForwardProp: buttonShouldForwardProp,
```

Comment:
How is this different than the implementation in the `experimentalStyled()`? If the difference is only the `classes`, let's again follow how it is done in the `Button` component - https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/Button/Button.js#L59

Reply:
Oh, but LoadingButton is in lab and it is not possible to import like this right?
`import { rootShouldForwardProp } from '@material-ui/core/styles/experimentalStyled';`

should it be exported from `core/styles` or add a TODO to fix the function once `LoadingButton` is moved to core?

Reply:
Let's not exported it at this moment. Let's just leave it as is...

Reply:
TODO added